sdk/src/types/block/capabilities.rs

```diff
@@ -21,11 +23,17 @@ pub struct Capabilities<Flag> {
 }
 
 impl<Flag> Capabilities<Flag> {
-    pub(crate) fn from_bytes(bytes: BoxedSlicePrefix<u8, u8>) -> Self {
-        Self {
+    /// Try to create capabilities from serialized bytes. Bytes with trailing zeroes are invalid.
+    pub(crate) fn from_bytes(bytes: BoxedSlicePrefix<u8, u8>) -> Result<Self, Error> {
+        if let Some(idx) = bytes.iter().rposition(|c| 0.ne(c)) {
+            if idx + 1 < bytes.len() {
```

Comment:
Not too sure about that, it could find a non-trailing 0 byte

Reply:
It's searching for `v != 0` from the right

Reply:
Let's say we have `[5, 12, 30, 5, 0]`, length is 5, idx found is 3 and 3 + 1 < 5 ?
Or am I too tired for this

Reply:
yes so that would be an error

Reply:
Yeah sorry I completely brainfarted.
Maybe rework into using `slice::last` would make it easier to understand.

Reply:
it has to be this way because there can be multiple trailing zeroes

Reply:
Sure but multiple always include the last one, no need to check more than that

Reply:
Hm...that's a really good point 😆